src/main/java/com/google/devtools/build/lib/actions/PathStripper.java

```diff
@@ -163,6 +215,60 @@ public interface CommandAdjuster {
      */
     List<String> stripCustomStarlarkArgs(List<String> args);
 
+    /**
+     * Creates a an {@link ArtifactPathMapper} that can be set on a Starlark thread's semantics
+     * object to automatically rewrite paths returned by the File API in map_each callbacks.
+     */
+    default ArtifactPathMapper toArtifactPathMapper() {
```

Comment:
I'm reading three meaningful abstractions now: `ArtifactPathMapper`, `PathMapper`, and `PathStripper`. I'm getting confused especially between the first two (sometimes I didn't understand a piece of code because I thought it was using a `PathMapper` vs. an `ArtifactPathMapper` or vice versa).

I'm not sure what the right answer is but I'd like to keep our abstractions as clear as we possibly can.

I'm also wondering if there's some way to better merge `PathMapper` and `PathStripper`. i.e. the latter is an instance of the former? And maybe avoid the current connections where `PathStripper` is broken down into distinct subcomponents `ActionStager` and `CommandAdjuster` and `PathMapper` re-combines them.

The idea behind the distinct subcomponents was distinct pieces of code were referencing each. Code that constructs command lines is concerned with `CommandAdjuster`. Code that stages actions is concerned with `ActionStager`. The thought was keeping each abstraction clear & focused for its intended consumer would be helpful for the API.